core/src/main/java/org/apache/iceberg/V1Metadata.java

```diff
@@ -236,34 +235,38 @@ static Types.StructType dataFileSchema(Types.StructType partitionType) {
   }
 
   /** Wrapper used to write a ManifestEntry to v1 metadata. */
-  static class IndexedManifestEntry implements ManifestEntry<DataFile>, IndexedRecord {
-    private final org.apache.avro.Schema avroSchema;
-    private final IndexedDataFile fileWrapper;
+  static class ManifestEntryWrapper implements ManifestEntry<DataFile>, StructLike {
+    private final int size;
+    private final DataFileWrapper fileWrapper;
     private ManifestEntry<DataFile> wrapped = null;
 
-    IndexedManifestEntry(Types.StructType partitionType) {
-      this.avroSchema = AvroSchemaUtil.convert(entrySchema(partitionType), "manifest_entry");
-      this.fileWrapper = new IndexedDataFile(avroSchema.getField("data_file").schema());
+    ManifestEntryWrapper() {
+      this.size = entrySchema(Types.StructType.of()).columns().size();
```

Comment:
nit: We have an EmptyStructLike we could use here.  Although maybe we should just have "of()" always return a singleton.